viewing_party/party.py

```diff
@@ -1,23 +1,213 @@
+    for user_movie_info in list_of_user_movie_info:
+        list_of_user_movie_titles.append(user_movie_info["title"])
+
+    #Adds friends' watched movie titles to a list
+    for frnds_watched_movie_info in list_of_frnds_watched_dicts:
+        for frnds_watched_movie_titles in frnds_watched_movie_info["watched"]:
+            list_of_friends_movie_titles.append(frnds_watched_movie_titles["title"])
+    
+    #Uses sets data structure and difference method to add friends' unique move titles that at least one friend has seen to a list
+    list_of_unique_frnds_movie_titles=set(list_of_friends_movie_titles).difference(set(list_of_user_movie_titles))
+    
+    #Adds movie info from a list of friends' watched movie info to a new list if the movie title is in the list of friends' unique movie titles (and isn't already in the new list)
+    for frnds_watched_movie_info in list_of_frnds_watched_dicts:
+        for frnds_movie_info in frnds_watched_movie_info["watched"]:
+            if frnds_movie_info["title"] in list_of_unique_frnds_movie_titles and frnds_movie_info not in list_of_unique_frnds_movie_info:
+                list_of_unique_frnds_movie_info.append(frnds_movie_info)
+    return list_of_unique_frnds_movie_info
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(single_user_data):
+    list_of_user_movie_info=single_user_data["watched"]
+    list_of_user_movie_hosts=single_user_data["subscriptions"]
+    list_of_users_unwatched_movie_info=[]
+    list_of_recommended_movie_info=[]
+
+    #Uses wave_03 function to add user's friends' unique watched movie info to a list
+    list_of_unique_frnds_movie_info=get_friends_unique_watched(single_user_data)
+
+    #Adds friends' movie info to user's unwatched list if it has not already been watched
+    for movie_info in list_of_unique_frnds_movie_info:
+        if movie_info not in list_of_user_movie_info:
+            list_of_users_unwatched_movie_info.append(movie_info)
+
+    #Adds user's unwatched movie into to user's recommended list if they are subscribed to the movie's host
+    for movie_info in list_of_users_unwatched_movie_info:
+        if movie_info["host"] in list_of_user_movie_hosts:
+            list_of_recommended_movie_info.append(movie_info)
+    return list_of_recommended_movie_info
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(single_user_data):
+    #Returns an empty list if the user does not have a most frequent genre
+    if len(single_user_data["watched"]) == 0:
+        return []
+    
+    #Returns an empty list if the user's friends do not have watchlists with movie info that can be used to get available movie recommendations
+    if len(get_friends_unique_watched(single_user_data)) == 0 :
+        return []

```

Comment:
great job putting these edge case checks at the top (sometimes called "guard clauses")